lib/slosilo/key.rb

```diff
@@ -121,6 +121,12 @@ def stringify value
         value
       else
         value.to_json
+      end.tap do |string|
+        # Ensure that the string is encoded as raw bytes so that we don't have
+        # encoding errors when smushing it together with the salt.
+        if string.encoding != Encoding::ASCII_8BIT
+          string.force_encoding Encoding::ASCII_8BIT
+        end
```

Comment:
Looks good, thanks! Some minor qualms: 
- `if` is redundant since `force_encoding` is a noop if it doesn't change anything (but see below),
- it modifies the original if it does change the encoding,
- it doesn't enforce a normalized string encoding (which of course is part of the original problem, but it's not entirely solved with this change).

I suggest:
- not touching binary strings (by keeping the `if`, or `unless` for clarity),
- for text, ensuring UTF-8 encoding (by using something like `return string.encode(Encoding::UTF_8).force_encoding(Encoding::ASCII_8BIT)`, which also copies the string, thus avoiding modifying the original).

Great catch, @kstutsman-conjur!